### PR TITLE
Feature/migrate classes to kotlin

### LINKED ISSUE #4: Migrate classes to Kotlin
In the project we have two java classes:

- EgnValidator.java
- EmbgValidator.java

These are the only java classes and they should be migrated to Kotlin.

### PR DESCRIPTION
**Pull request by mmdollar**
Fixes #4

To test: EGN and EMBG validators.

Note: I have migrated the classes, but I decided not to do a refactor. I prefer the refactor part to be another issue.

PR submission checklist:

- [x] I have followed the [Contributing Guidelines](https://github.com/scalefocus/virusafe-android/blob/master/CONTRIBUTING.md)
- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes and have described them above.

### CODE REVIEW

**Review by IvanYuliyanov — COMMENTED** [review]
It seems that you are using an old version of CI.yml file and may be this is the reason the build is failed. Please update it with the last version and try the build again.

**Comment by mmdollar** [comment]
> It seems that you are using an old version of CI.yml file and may be this is the reason the build is failed. Please update it with the last version and try the build again.

I did, but the problem still persists ;/

File: presentation/src/main/java/bg/government/virusafe/app/utils/validators/EgnValidator.kt [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	override fun clearAll() {
+		personalNumber = null
+		isValid = false
+		years = null
+		gender = null
+	}
+
+	@Suppress("ReturnCount")
+	private fun isValidPersonalNumber(): Boolean {
+		if (personalNumber == null || personalNumber == String.empty) {
```
Comment (IvanYuliyanov) [inline presentation/src/main/java/bg/government/virusafe/app/utils/validators/EgnValidator.kt]: Here you can use isNullOrEmpty() method

File: presentation/src/main/java/bg/government/virusafe/app/utils/validators/EgnValidator.kt [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+			return true
+		} else if (personalNumber!!.length != 10) {
```
Comment (IvanYuliyanov) [inline presentation/src/main/java/bg/government/virusafe/app/utils/validators/EgnValidator.kt]: Avoid using the not-null assertion operator (!!) use safe call operator instead (?.)

File: presentation/src/main/java/bg/government/virusafe/app/utils/validators/EmbgValidator.kt [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	override fun clearAll() {
+		personalNumber = null
+		isValid = false
+		years = null
+		gender = null
+	}
+
+	@Suppress("ReturnCount")
+	private fun isValidPersonalNumber(): Boolean {
+		if (personalNumber == null || personalNumber == String.empty) {
```
Comment (IvanYuliyanov) [inline presentation/src/main/java/bg/government/virusafe/app/utils/validators/EmbgValidator.kt]: Here you can use isNullOrEmpty() method

File: presentation/src/main/java/bg/government/virusafe/app/utils/validators/EmbgValidator.kt [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+			return true
+		} else if (personalNumber!!.length != 13) {
```
Comment (IvanYuliyanov) [inline presentation/src/main/java/bg/government/virusafe/app/utils/validators/EmbgValidator.kt]: Avoid using the not-null assertion operator (!!) use safe call operator instead (?.)

File: presentation/src/main/java/bg/government/virusafe/app/utils/validators/EgnValidator.kt [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+	@Suppress("ReturnCount")
+	private fun isValidPersonalNumber(): Boolean {
+		if (personalNumber.isNullOrEmpty()) {
+			return true
+		} else if (personalNumber?.length != 10) {
+			return false
+		}
+
+		val charArr = personalNumber!!.toCharArray()
```
Comment (IvanYuliyanov) [inline presentation/src/main/java/bg/government/virusafe/app/utils/validators/EgnValidator.kt]: Not-null assertion operator is still present.

File: presentation/src/main/java/bg/government/virusafe/app/utils/validators/EmbgValidator.kt [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+	@Suppress("ReturnCount")
+	private fun isValidPersonalNumber(): Boolean {
+		if (personalNumber.isNullOrEmpty()) {
+			return true
+		} else if (personalNumber?.length != 13) {
+			return false
+		}
+
+		val charArr = personalNumber!!.toCharArray()
```
Comment (IvanYuliyanov) [inline presentation/src/main/java/bg/government/virusafe/app/utils/validators/EmbgValidator.kt]: Same here.

**Pull request closed without merging**